### DISCUSSION TITLE
How to import existing OneDrive Client profile into OneDriveGUI?

### DISCUSSION THREAD

**Discussion by WB-Student** [bpozdena/OneDriveGUI#144] [Q&A]
Hello @bpozdena,

Thank you for your work on the OneDriveGUI, it is a very valuable tool. I recently switched to Linux (almost-) full time and just installed @abraunegg's [OneDrive Client for Linux](https://github.com/abraunegg/onedrive). After setting that up, I downloaded the OneDriveGUI Appimage and ran it. In setting up the GUI, I tried to import a profile as I already have a OneDrive account and had already activated and ran the OneDrive Client for Linux. However, none of the profile names or config file paths I entered would allow the "Import" button to be clicked. 

Eventually, I just selected the "Create new OneDrive profile" option, created a new profile name, and went through the URL authorization step to complete set up. However, this led to a new OneDrive folder being created in my Home directory, so now I have two separate OneDrive folders on my system linked to my personal OneDrive account: One from my OneDrive Client for Linux setup and one from the OneDriveGUI setup. 

Can you tell me how I should have imported my OneDrive profile so that the GUI would reflect or correspond to my OneDrive Client for Linux file path? 

Is there a way that I can determine what "profile name" the OneDrive Client for Linux is using? 

Or, if the profile name refers specifically to a _OneDriveGUI profile_ and not _my general OneDrive account_, is there a way I can direct my OneDriveGUI profile to my OneDrive Client for Linux folder?

I would greatly benefit from your responses to these questions and appreciate any response you provide. 

Thank you very much.

**Comment by bpozdena** [comment]
Hi @WB-Student,
That's interesting. I remember testing this with filepahs containing characters from many languages, but it's possible I missed something. 

Or more likely the import wizard was not intuitive enough to follow... 

Could you please share a screenshot of the problem from the wizard? You can open it again by clicking on the person icon in the GUI , then `add profile`, `import...`

In the meantime, you can adjust the path to your original onedrive profile in `~/.config/onedrive-gui/profiles`

**Reply by WB-Student** [reply]
Hi @bpozdena,

Thank you for the quick response. Here are the screenshots you requested:

This is what I initially tried when attempting to import my OneDrive profile
![GUI Wizard Import](https://github.com/bpozdena/OneDriveGUI/assets/34949247/ef0c40ce-0f41-4138-b253-b8b966069e66)
You can see the Import button is not able to be clicked. 

I also tried to change the Sync Directory in the OneDriveGUI to point to my OneDrive Client for Linux directory:

Here is the default for "new" OneDriveGUI profile I created
![GUI Sync Directory](https://github.com/bpozdena/OneDriveGUI/assets/34949247/91879f69-9b0c-4346-b041-ff6871cd243b)

Here is when I tried to change it to my original OneDrive Client for Linux directory:
![GUI Sync Directory to OneDrive Client folder](https://github.com/bpozdena/OneDriveGUI/assets/34949247/aad9756f-513a-4d4c-ac3a-120daf8e6fc3)
You can see I get a warning sign next to my profile name (I did not save this change). 

Here is what my home directory looks like, showing both the OneDrive Client for Linux directory (in red) and the OneDriveGUI directory (in blue). 
![Home Directory](https://github.com/bpozdena/OneDriveGUI/assets/34949247/cdfb4b30-1307-46be-b9df-637168a92f66)

And if it is helpful, here is my OneDrive Client for Linux config:
![OneDrive Client Config](https://github.com/bpozdena/OneDriveGUI/assets/34949247/766195f3-8b2d-4c5e-99d4-93dea16055db)

and my profile I created for the GUI:
![image](https://github.com/bpozdena/OneDriveGUI/assets/34949247/d36658b7-9459-4217-87db-2388d0023f2f)

**Reply by bpozdena** [reply]
Thanks for the details, @WB-Student . 

It seems you mixed up the onedrive sync dir  with config dir `-)  

![image](https://github.com/bpozdena/OneDriveGUI/assets/24818591/f0e1a5f2-bec7-4e6b-bedf-03a8aa7a4dcb)

There should also be a hint when you empty the text field:
![image](https://github.com/bpozdena/OneDriveGUI/assets/24818591/216dd296-c918-4039-8769-8c90e9fe5af1)



I would recommend you to delete the entire directory `~/.config/onedrive-gui/`. This will simply remove all GUI configuration.  Then start the GUI and import your original profile as shown on the above screenshot. 

Once you are happy with how everything works, you can delete your unwanted sync dir `~/OneDrive_william-b...` and config dir `~/.config/onedrive/accounts/`.

**Reply by WB-Student** [reply]
Hi @bpozdena,

That worked! Check it out:
![GUI Import](https://github.com/bpozdena/OneDriveGUI/assets/34949247/1dd1f01b-1764-4c3b-ba34-a863ad683e42)
![Sync Directory](https://github.com/bpozdena/OneDriveGUI/assets/34949247/04c0f918-ba08-4321-9ec2-452c41c7e981)

So to recap for anyone else to avoid my mistakes:
1. The "profile name" is specific to the OneDriveGUI itself and separate or independent of an individual's actual Microsoft account name.
2. Importing a profile requires pointing the OneDriveGUI to a config file. In my case, I was using all of the defaults in @abraunegg's [OneDrive Client for Linux](https://github.com/abraunegg/onedrive). As such, there was no config file in my OneDrive Client for Linux directory. After downloading the default config file ([here](https://github.com/abraunegg/onedrive/blob/master/config)) and putting it in my OneDrive Client for Linux config directory (/home/will/.config/onedrive), I could point the OneDriveGUI to this as a my profile config path and successfully "import" my profile. 
3. I could set my original OneDrive Client for Linux as my OneDriveGUI sync directory and get rid of the duplicate I erroneously created during my first GUI setup where I created a new profile. 

I have tried all of this out and am happy to say everything is working smoothly. 

Thank you so much for you time, patience, and guidance, @bpozdena.

**Reply by gb5757870** [reply]
Thank you for sharing!!

**Comment by galettof** [comment]
Hi, a related question. What if there is no config file at all? My understanding from the onedrive client documentation is that in the default configuration no config file is created, which makes it seemingly impossible to import a profile into the GUI. Any suggestions? Thanks.

**Comment by bpozdena** [comment]
I never envisioned a scenario where someone would want to import existing profile without a config. 
Manually creating an empty 'config' file should do the trick for now.

**Reply by galettof** [reply]
Thanks for the quick reply.

Do you actually mean an empty text file, or should I create one with the default config from the onedrive client?

**Comment by bpozdena** [comment]
Just am empty text file called 'config' should be fine. You'll then select it in the GUI during the import.

**Reply by galettof** [reply]
Perfect, I'll give it a try soon.
